File: src/main/java/br/com/blu/bs2/sintetizador/contas/csv/model/enums/TipoExtensaoArquivo.java
package br.com.blu.bs2.sintetizador.contas.csv.model.enums;

import java.util.EnumSet;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public enum TipoExtensaoArquivo {

    CSV(1, "CSV", "TIPO EXTENSAO ARQUIVO CSV"),

    TXT(2, "TXT", "TIPO EXTENSAO ARQUIVO TXT"),

    DAT(3, "DAT", "TIPO EXTENSAO ARQUIVO DAT");

    private static final Map<Integer, TipoExtensaoArquivo> lookup;

    static {
        lookup = new HashMap<>();
        EnumSet<TipoExtensaoArquivo> enumSet = EnumSet.allOf(TipoExtensaoArquivo.class);

        for (TipoExtensaoArquivo type : enumSet)
            lookup.put(type.codigo, type);
    }

    private int codigo;
    private String codigoLiteral;
    private String descricao;

    TipoExtensaoArquivo(int codigo, String codigoLiteral, String descricao) {
        inicialize(codigo, codigoLiteral, descricao);
    }

    private void inicialize(int codigo, String codigoLiteral, String descricao) {
        this.codigo = codigo;
        this.codigoLiteral = codigoLiteral;
        this.descricao = descricao;
    }

    public static TipoExtensaoArquivo fromCodigo(int codigo) {
        if (lookup.containsKey(codigo))
            return lookup.get(codigo);
        throw new IllegalArgumentException(String.format("Código do Layout Arquivo inválido: %d", codigo));
    }

    public int getCodigo() {
        return this.codigo;
    }

    public String getCodigoLiteral() {
        return this.codigoLiteral.toLowerCase();
    }

    public String getDescricao() {
        return this.descricao;
    }

    public boolean isXML() {
        return Objects.equals(this, CSV);
    }

    public boolean isTXT() {
        return Objects.equals(this, TXT);
    }

    public boolean isDAT() {
        return Objects.equals(this, DAT);
    }
}

File: src/main/kotlin/br/com/blu/bs2/sintetizador/contas/csv/model/enums/TipoExtensaoArquivo.kt
package br.com.blu.bs2.sintetizador.contas.csv.model.enums

enum class TipoExtensaoArquivo(codigo: Int, codigoLiteral: String, descricao: String) {
    CSV(1, "CSV", "TIPO EXTENSAO ARQUIVO CSV"), TXT(2, "TXT", "TIPO EXTENSAO ARQUIVO TXT"), DAT(3, "DAT", "TIPO EXTENSAO ARQUIVO DAT");

    private var codigo = 0
    private var codigoLiteral: String? = null
    private var descricao: String? = null

    private fun inicialize(codigo: Int, codigoLiteral: String, descricao: String) {
        this.codigo = codigo
        this.codigoLiteral = codigoLiteral
        this.descricao = descricao
    }

    fun getCodigoLiteral(): String {
        return codigoLiteral!!.toLowerCase()
    }

    val isXML: Boolean
        get() = this == CSV
    val isTXT: Boolean
        get() = this == TXT
    val isDAT: Boolean
        get() = this == DAT

    companion object {
        @Throws(IllegalArgumentException::class)
        fun of(codigo: Int): TipoExtensaoArquivo {
            for (tipoExtensaoArquivo in values()) {
                if (tipoExtensaoArquivo.codigo == codigo)
                    return tipoExtensaoArquivo
            }
            throw IllegalArgumentException(String.format("Código do Tipo Extensão Arquivo inválido: %d", codigo))
        }
    }

    init {
        inicialize(codigo, codigoLiteral, descricao)
    }
}
